src/WingetCreateCLI/Commands/BaseCommand.cs

```diff
@@ -256,6 +257,26 @@ protected static bool ValidateManifestsInTempDir(Manifests manifests)
             return result;
         }
 
+        /// <summary>
+        /// Displays the appropriate warning messages for installers with detected architecture mismatches.
+        /// </summary>
+        /// <param name="detectedArchs">List of DetectedArch objects that represent each installers detected architectures.</param>
+        protected static void DisplayMismatchedArchitectures(List<PackageParser.DetectedArch> detectedArchs)
+        {
+            var mismatchedArchInstallers = detectedArchs.Where(i => i.UrlArch != i.BinaryArch).ToList();
+            if (mismatchedArchInstallers.Any())
+            {
+                Logger.WarnLocalized(nameof(Resources.DetectedArchMismatch_Message));
+                Console.WriteLine();
+                mismatchedArchInstallers.ForEach(i =>
```

Comment:
nit: I'd use foreach syntax here, it's more common. You also don't need the .ToList() if you do that

Reply:
changed to foreach syntax